.github/workflows/main.yml

```diff
@@ -0,0 +1,66 @@
+name: lf-api-client-core-java-CI
+
+on:
+  push:
+    branches: [ '\d+.x' ]
+  pull_request:
+    branches: [ '\d+.x' ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  VERSION_PREFIX: 1.0.0
+
+jobs:
+  build-n-test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Set up JDK 8
+      uses: actions/setup-java@v3
+      with:
+        java-version: '8'
+        distribution: 'temurin'
+        
+    - name: Build with Maven
+      run: mvn -B package -Dmaven.test.skip=true --file pom.xml
+      
+    - name: Run tests
+      env:
+        ACCESS_KEY: ${{ secrets.DEV_CA_PUBLIC_USE_INTEGRATION_TEST_ACCESS_KEY }}
+        SERVICE_PRINCIPAL_KEY: ${{ secrets.DEV_CA_PUBLIC_USE_TESTOAUTHSERVICEPRINCIPAL_SERVICE_PRINCIPAL_KEY }}
+      run: mvn test
+
+  build-documentation:
+
+    runs-on: ubuntu-latest
+    needs: [ build-n-test ]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up JDK 8
+      uses: actions/setup-java@v3
+      with:
+        java-version: '8'
+        distribution: 'temurin'
+
+    - name: Generate Javadoc
+      run: mvn javadoc:javadoc
+
+    - name: Upload a Build Artifact
+      uses: actions/upload-artifact@v2.3.1
+      with:
+        name: documentation-artifact
+        # A file, directory or wildcard pattern that describes what to upload
+        path: ${{ github.workspace }}/target/site/
```

Comment:
Do we save it on GitHub? Is there a limit how much we can store?

Reply:
i just tried to mirrored the steps on the other 2 projects, in terms of the limit, i don't think there is any limit specifically relating the max size of the artifact that you can upload after trying to search it up although there are a bunch of other limits like limit for git push file sizes, cache limits, git hub repo size, etc.

Reply:
Here is what @samadpaydar3 did:
https://github.com/Laserfiche/lf-api-client-core-dotnet/blob/1.x/.github/workflows/main.yml#L80

Reply:
In short, the generated doc was stored in a directory with "temp" in its name. I think this might be a good idea since it tells that the storage is temporary. It then deletes the file.

Reply:
I think we could follow Samad's existing code.

Reply:
ok done